internal/resto/client.go

```diff
@@ -0,0 +1,115 @@
+package resto
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const (
+	completeJobStatus string = "complete"
+	errorJobStatus    string = "error"
+)
+
+var jobStatuses = map[string]struct{}{
+	completeJobStatus: struct{}{},
+	errorJobStatus:    struct{}{},
+}
+
+var (
+	// ErrServerInaccessible represents error message when server is inaccessible.
+	ErrServerInaccessible = errors.New("couldn't reach resto server")
+	// ErrJobNotFound represents error message from server when a job was not found.
+	ErrJobNotFound = errors.New("job was not found")
+)
+
+// Client http client.
+type Client struct {
+	HTTPClient *http.Client
+	URL        string
+	Username   string
+	AccessKey  string
+}
+
+// New creates a new client.
+func New(url, username, accessKey string, timeout int) Client {
```

Comment:
On a second glance, I'd also just use `time.Duration` for the timeout. No real need for the mapping later on 👍 

```suggestion
func New(url, username, accessKey string, timeout time.Duration) Client {
```

Reply:
Let's change to time.Duration instead.